gitclone/checkout_method_pr_auto_merge_branch.go

```diff
@@ -40,8 +40,15 @@ func (c checkoutPRMergeRef) do(gitCmd git.Git, fetchOpts fetchOptions, fallback
 	// https://git-scm.com/book/en/v2/Git-Internals-The-Refspec
 	refSpec := fmt.Sprintf("%s:%s", c.remoteMergeRef(), c.localMergeRef())
 
+	// delete remote branch ref
+	// $ git update-ref -d refs/remotes/pull/7/merge
+	err := deleteRef(gitCmd, c.remoteMergeRef())
+	if err != nil {
+		return fmt.Errorf("failed to delte ref: %w", err)
```

Comment:
The doc says that it first verifies that the ref exists before deleting it.. Which means we can't just run it all the time, it's going to bubble up the error if there is nothing to clean up.

Reply:
Tested on macos/linux, the git command does not fail.